### PR TITLE
update telemetry key for keyvault keys

### PR DESCRIPTION
**Pull request by vhvb1989**
fixes: https://github.com/Azure/azure-sdk-for-cpp/issues/2459

### CODE REVIEW

**Review by gearama — APPROVED** [review]

**Review by RickWinter — APPROVED** [review]

File: sdk/keyvault/azure-security-keyvault-keys/src/key_client.cpp [hunk 1]
```diff
@@ -23,6 +23,8 @@ using namespace Azure::Core::Http::Policies;
 using namespace Azure::Core::Http::Policies::_internal;
 
 namespace {
+constexpr static const char KeyVaultServicePackageName[] = "keyvault-keys";
```
Comment (ahsonkhan) [inline sdk/keyvault/azure-security-keyvault-keys/src/key_client.cpp:26]: Do other languages avoid using the full name "azure-security-keyvault-keys" for telemetry? What is the exact `service name`?

And should it be all lower-case?

@antkmsft, should we lower-case the `i` here?
https://github.com/Azure/azure-sdk-for-cpp/blob/fda88bf7b8389575b9d973821efc604442e10493/sdk/identity/azure-identity/src/client_secret_credential.cpp#L97

Comment (ahsonkhan) [reply sdk/keyvault/azure-security-keyvault-keys/src/key_client.cpp:26]: For reference, this is what storage does:

https://github.com/Azure/azure-sdk-for-cpp/blob/fda88bf7b8389575b9d973821efc604442e10493/sdk/storage/azure-storage-common/inc/azure/storage/common/internal/constants.hpp#L7-L10

Comment (ahsonkhan) [reply sdk/keyvault/azure-security-keyvault-keys/src/key_client.cpp:26]: @heaths what is the telemetry string for KeyVault Keys in .NET?
https://github.com/Azure/azure-sdk-for-net/blob/4162f6fa2445b2127468b9cfd080f01c9da88eba/sdk/core/Azure.Core/src/Pipeline/HttpPipelineBuilder.cs#L106-L118

**Review by ahsonkhan — APPROVED** [review]
Confirm what other languages do when it comes to keyvault-keys and what is considered the "service name"/"package name". I am wondering if it should be "azure-security-keyvault-keys".

**Pull request merged**